docs/src/simulation.md

```diff
@@ -0,0 +1,183 @@
+Simulating an Experiment from Scratch
+======================================
+
+Here is a worked example of simulating a partially crossed design from scratch.
+
+First, some setup:
+
+```@example Main
+using DataFrames
+using LinearAlgebra
+using MixedModels, MixedModelsSim
+using Random
+```
+
+## Assemble the Design
+
+We're going to do a 2 x 2 x 2 design.
+For concreteness, let's think of this as a linguistic design:
+
+- **age** `old` vs. `young`, _between subjects_
+- **frequency** `high` vs. `low`, _between items_
+- **context** `matched` vs. `unmatched`, _within both_.
+
+Further, let's assume we want 40 subjects and 80 items.
+We can specify this design as follows:
+
+```@example Main
+n_subj = 40
+n_item = 80
+subj_btwn = Dict(:age => ["old", "young"])
+item_btwn = Dict(:frequency => ["high", "low"])
+both_win = Dict(:context => ["matched", "unmatched"])
+```
+
+and then generate it:
+
+```@example Main
+rng = MersenneTwister(42)  # specify our random number generator for reproducibility
+design = simdat_crossed(rng, n_subj, n_item;
+                        subj_btwn = subj_btwn,
+                        item_btwn = item_btwn,
+                        both_win = both_win)
+```
+
+Note that `simdat_crossed` returns a row table, which `MixedModels.jl` can process directly.
+For nicely displaying it, we can also convert it to a dataframe and change the factors to use pooled arrays to save a bit of memory.
+
+```@example Main
+df = pooled!(DataFrame(design))
+first(df, 5)
+```
+
+Note that `simdat_crossed` generated a column `dv` for our dependent variable that has been pre-populated with noise from a standard normal distribution ($N(0,1)$).
+Typically, we will want to scale that, but we can do that in the simulation step.
+Also, this dependent variable is *pure noise*: we haven't yet added in effects.
+Adding in effects also comes in the simulation step.
+
+But before we get to simulating, let's fit the model to the noise, just to see how things look. We're going to use effects coding for our contrasts.
+
+```@example Main
+contrasts = Dict(:age => EffectsCoding(base="young"),
+                 :frequency => EffectsCoding(base="high"),
+                 :context => EffectsCoding(base="matched"))
+form = @formula(dv ~ 1 + age * frequency * context +
```

Comment:
Not to get into contrast wars but I prefer +/- 1 encoding of two-level factors, such as you get from `HelmertCoding`.

Reply:
:smile: In this case, it works out to be the same. I went with `EffectsCoding` because that seems to be the more popular in psycholinguistics (to the extent that people don't either (1) use the default contrasts or (2)  use very careful custom hypothesis matrices. 

I'm not adverse to changing this to Helmert. The most important aspect for me was to have the contrasts stated explicitly in order to set a good example. (And set up space for a later update discussing the interpretation of the coefficients in the simulation.)

Reply:
That's fine.  I had forgotten that `EffectsCoding` produced +/- 1 contrasts.